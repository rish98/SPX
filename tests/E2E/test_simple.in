[Products; GPU ,Router]

===== Trader 0 =====
[Trader 0] [t=0] Received from SPX: MARKET OPEN;
[Trader 0] [t=1] Event: BUY 0 GPU 10 20
[Trader 0] [t=1] Received from SPX: ACCEPTED 0;
[Trader 0] [t=2] Event: DISCONNECT

===== Trader 1 =====
[Trader 0] [t=0] Received from SPX: MARKET OPEN;
[Trader 0] [t=1] Event: SELL 0 Router 15 800
[Trader 0] [t=1] Received from SPX: ACCEPTED 0;
[Trader 0] [t=2] Event: DISCONNECT


