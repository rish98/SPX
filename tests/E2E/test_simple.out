===== Exchange STDOUT =====
[SPX] Starting
[SPX] Trading 2 products: GPU Router
[SPX] Created FIFO /tmp/spx_exchange_0
[SPX] Created FIFO /tmp/spx_trader_0
[SPX] Starting trader 0 (./bin/spx_test_trader)
[SPX] Connected to /tmp/spx_exchange_0
[SPX] Connected to /tmp/spx_trader_0
[SPX] Created FIFO /tmp/spx_exchange_1
[SPX] Created FIFO /tmp/spx_trader_1
[SPX] Starting trader 1 (./bin/spx_test_trader)
[SPX] Connected to /tmp/spx_exchange_1
[SPX] Connected to /tmp/spx_trader_1
[SPX] [T0] Parsing command: <BUY 0 GPU 10 20>
[SPX]     --ORDERBOOK--
[SPX]     Product: GPU; Buy levels: 1; Sell levels: 0
[SPX]          BUY 10 @ $20 (1 order)
[SPX]     Product: Router; Buy levels: 0; Sell levels: 0
[SPX]     --POSITIONS--
[SPX]     Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX] [T1] Parsing command: <SELL 0 Router 15 800>
[SPX]     --ORDERBOOK--
[SPX]     Product: GPU; Buy levels: 1; Sell levels: 0
[SPX]          BUY 10 @ $20 (1 order)
[SPX]     Product: Router; Buy levels: 0; Sell levels: 1
[SPX]          SELL 15 @ $800 (1 order)
[SPX]     --POSITIONS--
[SPX]     Trader 0: GPU 0 ($0), Router 0 ($0)
[SPX] Trader 0 disconnected
[SPX] Trader 1 disconnected
[SPX] Trading completed
[SPX] Exchange fees collected: $0